Add URL Namer plugin

zfei opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/zfei/obsidian-url-namer

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [ ] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
You can remove `obsidian` from your plugin id, its unnecesarry.

[static rawUrlPattern = /(?<!\]\(\s*)(?<=\s|\(|\[|^)(?:https?:\/\/)?[a-zA-Z0-9]+[a-zA-Z0-9\-_.]*\.[a-z]{2,6}[^\s]*\b/gim;](https://github.com/zfei/obsidian-url-namer/blob/b4605217e1b65c3a1f933da50bb53f195d1051c0/main.ts#L50)
As you are using lookbehinds here this plugin will not work on iOS

You should add a notice while the plugin is doing its thing.
Just set the duration to `0` and call the `hide()` method on the notice once its finished.

[const res = await requestUrl({ url: reqUrl });](https://github.com/zfei/obsidian-url-namer/blob/b4605217e1b65c3a1f933da50bb53f195d1051c0/main.ts#L105)
You should check `res.status` if the request was succesfull, it returns HTTP status codes.

zfei commented:
@joethei thanks for the kind comments.

As I'm looking to parametrize the url pattern regex into plugin settings, will just leave the compatibility flag as is.

Removed the `obsidian-` prefix in the plugin json. Addressed the comments on the loading indicator notice and response status code check in the latest commit: https://github.com/zfei/obsidian-url-namer/commit/6cc1474e3c8fd365e52a4287ddbedaa315c7085d.

liamcain commented on community-plugins.json:
```diff
@@ -4987,5 +4987,13 @@
+        "name": "URL Namer",
+        "author": "zfei",
+        "description": "This plugin retrieves the HTML titles to name the raw URL links.",
+        "repo": "https://github.com/zfei/obsidian-url-namer",
```
```suggestion
        "repo": "zfei/obsidian-url-namer",
```
